### PR TITLE
Cleanup of raw pointers in CNodePairInfo

### PR DESCRIPTION
**Pull request by cochicde**
Just removing some raw pointers and using `unique_ptr` or objects directly.

It makes memory handling safer and keep destructors empty.

The major change is in `class CNodePairInfo`. The rest is refactoring the code to make it work

### CODE REVIEW

File: src/com/opc_ua/opcua_action_info.h [hunk 1]
```diff
@@ -46,11 +46,26 @@ class CActionInfo {
...
         CNodePairInfo(UA_NodeId *paNodeId, const std::string &paBrowsePath) :
-            mNodeId(paNodeId), mBrowsePath(paBrowsePath) {
+            mNodeId(paNodeId, UA_NodeId_delete), mBrowsePath(paBrowsePath) {
+        }
+
+        UA_NodeId *getNodeId() const {
+          return mNodeId.get();
+        }
+
+        void setNodeId(UA_NodeId * paNodeId){
```
Comment (m-meingast) [inline src/com/opc_ua/opcua_action_info.h]: There is a blank space between the asterisk and the parameter name.
```suggestion
        void setNodeId(UA_NodeId *paNodeId){
```

File: src/com/opc_ua/opcua_local_handler.cpp [hunk 2]
```diff
@@ -906,9 +908,9 @@ UA_StatusCode COPC_UA_Local_Handler::initializeCreateNode(CActionInfo &paActionI
   //The main process is done in the execution
   UA_StatusCode retVal = UA_STATUSCODE_GOOD;
 
-  CSinglyLinkedList<CActionInfo::CNodePairInfo*>::Iterator it = paActionInfo.getNodePairInfo().back();
+  auto& instance = paActionInfo.getNodePairInfo().back();
```
Comment (m-meingast) [inline src/com/opc_ua/opcua_local_handler.cpp]: Maybe we should use a more explicit variable name instead of instance, e.g. nodePair?

**Review by m-meingast — APPROVED** [review]
Looks good to me.

File: src/com/opc_ua/opcua_action_info.h (hunk 1 above)
Comment (cochicde) [reply src/com/opc_ua/opcua_action_info.h]: changed

File: src/com/opc_ua/opcua_local_handler.cpp (hunk 2 above)
Comment (cochicde) [reply src/com/opc_ua/opcua_local_handler.cpp]: you're right. Changed

**Review by m-meingast — APPROVED** [review]

**Review by azoitl — APPROVED** [review]

**Pull request merged**